pennylane_lightning/lightning_kokkos/lightning_kokkos.py

```diff
@@ -450,7 +613,13 @@ def supports_derivatives(
             Bool: Whether or not a derivative can be calculated provided the given information
 
         """
-        return 0
+        if execution_config is None and circuit is None:
+            return True
+        if execution_config.gradient_method not in {"adjoint", "best"}:
```

Comment:
Can't this be `None` too?
```suggestion
        if execution_config.gradient_method not in {None, "adjoint", "best"}:
```

Reply:
I tried this suggestion but some tests failed. I think this is because the `gradient_method` only has two options. However, if the variable `execution_config` is `None` then the circuit does not need to support derivatives. Moreover, we can propose a new condition which says if `gradient_method = None` becomes `gradient_method = best`

Reply:
I looked again at the code trying to understand your suggestion but it seems that `gradient_method` should only allocate the available gradient methods. If `gradient_method = None` in the subsequent steps will have problems because the `ExecutionConfig` class will not know which gradient method to call. :confused:

Reply:
This is how it is done in PL, never mind me :)